daemon/algod/api/server/v2/dryrun.go

```diff
@@ -182,23 +182,23 @@ func (ddr *dryrunDebugReceiver) stateToState(state *logic.DebugState) generated.
 }
 
 // Register is fired on program creation (DebuggerHook interface)
-func (ddr *dryrunDebugReceiver) Register(state *logic.DebugState) error {
+func (ddr *dryrunDebugReceiver) BeforeAppEval(state *logic.DebugState) error {
 	ddr.disassembly = state.Disassembly
 	ddr.lines = strings.Split(state.Disassembly, "\n")
 	return nil
 }
 
 // Update is fired on every step (DebuggerHook interface)
-func (ddr *dryrunDebugReceiver) Update(state *logic.DebugState) error {
+func (ddr *dryrunDebugReceiver) BeforeTealOp(state *logic.DebugState) error {
 	st := ddr.stateToState(state)
 	ddr.history = append(ddr.history, st)
 	ddr.updateScratch()
 	return nil
 }
 
 // Complete is called when the program exits (DebuggerHook interface)
-func (ddr *dryrunDebugReceiver) Complete(state *logic.DebugState) error {
-	return ddr.Update(state)
+func (ddr *dryrunDebugReceiver) AfterAppEval(state *logic.DebugState) error {
+	return ddr.AfterAppEval(state)
```

Comment:
This does look to be a recursive call. Is that a mistake?

Reply:
Yes, it most definitely is. Only a little concerning that the dryrun tests didn't catch it.